webui/src/lib/components/repository/changes.jsx

```diff
@@ -212,65 +119,30 @@ export const TreeEntryPaginator = ({ path, setAfterUpdated, nextPage, depth=0, l
     );
 };
 
-function extractPathText(entry, relativeTo) {
-    let pathText = entry.path;
-    if (pathText.startsWith(relativeTo)) {
-        pathText = pathText.substr(relativeTo.length);
-    }
-    return pathText;
-}
-
-function diffType(entry) {
-    switch (entry.type) {
-        case 'changed':
-        case 'prefix_changed':
-            return 'diff-changed';
-        case 'added':
-            return 'diff-added';
-        case 'removed':
-            return 'diff-removed';
-        case 'conflict':
-            return 'diff-conflict';
-        default:
-            return '';
-    }
-}
-
-function diffIndicatorIcon(entry) {
-    if (entry.path_type === 'common_prefix') {
-        return <OverlayTrigger placement="bottom" overlay={(<Tooltip id={"tooltip-prefix"}>Changes under prefix</Tooltip>)}>
-                        <span>
-                            <FileDirectoryIcon/>
-                        </span>
-               </OverlayTrigger>;
-    }
-
-    switch (entry.type) {
-        case 'removed':
-            return <OverlayTrigger placement="bottom" overlay={(<Tooltip id={"tooltip-removed"}>Removed</Tooltip>)}>
-                        <span>
-                            <TrashIcon/>
-                        </span>
-                    </OverlayTrigger>;
-        case 'added':
-            return <OverlayTrigger placement="bottom" overlay={(<Tooltip id={"tooltip-added"}>Added</Tooltip>)}>
-                        <span>
-                            <PlusIcon/>
-                        </span>
-                    </OverlayTrigger>;
-        case 'changed':
-            return <OverlayTrigger placement="bottom" overlay={(<Tooltip id={"tooltip-changed"}>Changed</Tooltip>)}>
-                        <span>
-                            <PencilIcon/>
-                        </span>
-                    </OverlayTrigger>;
-        case 'conflict':
-            return <OverlayTrigger placement="bottom" overlay={(<Tooltip id={"tooltip-conflict"}>Conflict</Tooltip>)}>
-                        <span>
-                            <CircleSlashIcon/>
-                        </span>
-                    </OverlayTrigger>;
-        default:
-            return '';
-    }
+function useTreeItemType(entry, repo, leftDiffRefID, rightDiffRefID) {
+    const [treeItemType, setTreeItemType] = useState({type: null, loading: true});
+
+    // Tree items that represent prefixes are always of entry.type = prefix_changed and the actual diff type is
+    // presented at the object level. Therefore, in case of tables that were added or removed we don't know
+    // under which of the diff refs the table root is expected to be listed and therefore we try to get the table type
+    // from both and take the one that returned results.
+    let leftResult = useAPI(() => tablesUtil.isDeltaLakeTable(entry, repo, rightDiffRefID));
```

Comment:
Consider not sending these requests if the feature is disabled.

Reply:
Changed it not to send the requests, thanks

Reply:
nit: might be better to move the check to this point and let the `isDeltaLakeTable` do a single thing.
(as mentioned - a nit)

Reply:
If you meant to the check whether deltadiff is enables, since we are going to remove the feature flag anyways i'm keeping it for now and going with the current version.